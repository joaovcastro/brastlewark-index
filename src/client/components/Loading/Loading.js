import React from 'react';
import styled from 'styled-components';

const LoadingWrapper = styled.div`
  display: flex;
  position: absolute;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  width: 100vw;
  background-color: #ededed;
  margin-top: -10px;
`;

const LoadingAnimation = styled.span`
  width: 35px;
  height: 35px;
  display: inline-block;
  border: 5px solid rgba(101, 131, 97, 0.25);
  border-left-color: rgba(101, 131, 97, 1);
  border-top-color: rgba(101, 131, 979, 1);
  border-radius: 50%;
  @keyframes rotate {
    to {
      transform: rotate(1turn);
    }
  }
  animation: rotate 1000ms infinite linear;
`;

const Loading = () => (
  <LoadingWrapper>
    <LoadingAnimation />
  </LoadingWrapper>
);

export default Loading;
