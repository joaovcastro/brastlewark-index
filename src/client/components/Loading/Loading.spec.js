import React from 'react';
import { shallow } from 'enzyme';

import Loading from './Loading';

describe('<Loading />', () => {
  it('correctly renders', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
