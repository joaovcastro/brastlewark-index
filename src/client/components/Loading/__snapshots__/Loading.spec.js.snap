// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> correctly renders 1`] = `ShallowWrapper {}`;
